cdap-app-fabric/src/main/java/co/cask/cdap/metadata/MetadataHttpHandler.java

```diff
@@ -521,23 +522,30 @@ public void removeStreamTags(HttpRequest request, HttpResponder responder,
   }
 
   private Map<String, String> readMetadata(HttpRequest request) throws BadRequestException, IOException {
-    ChannelBuffer content = request.getContent();
-    if (!content.readable()) {
-      throw new BadRequestException("Unable to read business metadata from request.");
-    }
-    try (Reader reader = new InputStreamReader(new ChannelBufferInputStream(content), Charsets.UTF_8)) {
-      return GSON.fromJson(reader, MAP_STRING_STRING_TYPE);
+    try {
+      ChannelBuffer content = request.getContent();
+      Preconditions.checkArgument(content.readable());
```

Comment:
I think the previous code to return the BadRequestException here makes more sense and its more explicit.